apps/root/src/hooks/useTransactionReceipt.ts

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import parseTransactionEventToTransactionReceipt from '@common/utils/transaction-history/transaction-receipt-parser';
+import useTransactionsHistory from './useTransactionsHistory';
+
+function useTransactionReceipt(txHash: string) {
+  const transactionsHistory = useTransactionsHistory();
+
+  const transactionReceipt = React.useMemo(() => {
+    const transactionEvent = transactionsHistory.events.find((event) => event.tx.txHash === txHash);
+    return parseTransactionEventToTransactionReceipt(transactionEvent);
+  }, [transactionsHistory, txHash]);
```

Comment:
you can pass transactionHistory.events since its the only thing we care about changing